packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1576,6 +1576,9 @@ class MaterialBasedCupertinoThemeData extends CupertinoThemeData {
   @override
   Color get scaffoldBackgroundColor => _cupertinoOverrideTheme.scaffoldBackgroundColor ?? _materialTheme.scaffoldBackgroundColor;
 
+  @override
+  Color get barBackgroundColor => _cupertinoOverrideTheme.barBackgroundColor ?? _materialTheme.bottomAppBarTheme.color ?? _materialTheme.bottomAppBarColor;
```

Comment:
I'm not so sure `bottomAppBarColor` is the correct color that we want to derive `barBackgroundColor` from.